app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt

```diff
@@ -52,30 +57,62 @@ constructor(private val db: FirebaseFirestore, private val auth: FirebaseAuth) :
     db.collection(collectionPath)
         .get()
         .addOnSuccessListener { querySnapshot ->
-          Log.d("UserRepositoryFirestore", "getAllUsers")
           val users =
               querySnapshot.documents.mapNotNull { document ->
-                Log.d("UserRepositoryFirestore", document.data.toString())
-                document.data?.let { deserializeUser(it) }
+                val userPublic = deserializeUser(document.data!!)
+                User(userPublic!!, null)
               }
           onSuccess(users)
         }
         .addOnFailureListener { onFailure(it) }
   }
 
   override fun addUser(user: User, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection(collectionPath)
-        .document(user.userId)
-        .set(serializeUser(user))
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { onFailure(it) }
+    if (user.details == null) {
+      onFailure(Exception("User details are required"))
+      return
+    }
+    db.collection(collectionPath).document(user.public.userId).get().addOnSuccessListener {
+      if (it.exists()) {
+        onSuccess()
+        return@addOnSuccessListener
+      } else {
+        db.collection(collectionPath)
+            .document(user.public.userId)
+            .set(user.public)
+            .addOnSuccessListener {
+              db.collection(collectionPath)
+                  .document(user.public.userId)
+                  .collection("private")
+                  .document("details")
+                  .set(user.details!!)
+                  .addOnSuccessListener { onSuccess() }
+                  .addOnFailureListener { onFailure(it) }
+            }
+            .addOnFailureListener { onFailure(it) }
+      }
+    }
   }
 
   override fun updateUser(user: User, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
+    Log.d("UserRepositoryFirestore", "updateUser: $user")
+    if (user.details == null) {
+      Log.d("UserRepositoryFirestore", "updateUser: user details are null, can't update")
```

Comment:
Since we throw an exception here , is it necessary to keep this log.d ?

Reply:
No I forgot to remove my debugging log